Source/WebKit/WebProcess/GPU/webrtc/LibWebRTCCodecs.cpp

```diff
@@ -421,30 +419,48 @@ void LibWebRTCCodecs::setDecoderFormatDescription(Decoder& decoder, std::span<co
     decoder.connection->send(Messages::LibWebRTCCodecsProxy::SetDecoderFormatDescription { decoder.identifier, data, width, height }, 0);
 }
 
-void LibWebRTCCodecs::sendFrameToDecode(Decoder& decoder, int64_t timeStamp, std::span<const uint8_t> data, uint16_t width, uint16_t height, Function<void(bool)>&& callback)
+void LibWebRTCCodecs::sendFrameToDecode(Decoder& decoder, int64_t timeStamp, std::span<const uint8_t> data, uint16_t width, uint16_t height, std::optional<FramePromise::Producer>&& producer)
 {
     if (decoder.type == WebCore::VideoCodecType::VP9 && (width || height))
         decoder.connection->send(Messages::LibWebRTCCodecsProxy::SetFrameSize { decoder.identifier, width, height }, 0);
 
-    decoder.connection->sendWithPromisedReply(Messages::LibWebRTCCodecsProxy::DecodeFrame { decoder.identifier, timeStamp, data }, 0)->whenSettled(workQueue(), [callback = WTFMove(callback)] (auto&& result) mutable {
-        callback(result ? result.value() : false);
+    decoder.connection->sendWithPromisedReply(Messages::LibWebRTCCodecsProxy::DecodeFrame { decoder.identifier, timeStamp, data }, 0)->whenSettled(workQueue(), [producer = WTFMove(producer)] (auto&& result) mutable {
```

Comment:
same here, we don't need the optional Producer, we can have sendFrameToDecode return a promise too and directly return the sendWithPromisedReply value with the PromiseConverter template

Reply:
Ditto.